packages/intrepid2/src/Orientation/Intrepid2_OrientationToolsDefCoeffMatrix_HCURL.hpp

```diff
@@ -210,48 +203,47 @@ getCoeffMatrix_HCURL(OutputViewType &output,
   // on the subcell spanning the same space as the bases \phi_j
 
   // Tangents t_j
-  ScalarViewType subcellTangents("subcellTangents", numSubcellBasis, subcellDim);
+  Kokkos::DynRankView<value_type,host_device_type> subcellTangents("subcellTangents", numSubcellBasis, subcellDim);
   auto degree = subcellBasis.getDegree();
-  BasisPtr<ExecutionSpace, ScalarType, ScalarType> basisPtr;
+  BasisPtr<host_device_type, value_type, value_type> basisPtr;
   if(subcellBaseKey == shards::Line<>::key) {
-    basisPtr = Teuchos::rcp(new Intrepid2::Basis_HVOL_LINE_Cn_FEM<ExecutionSpace, ScalarType, ScalarType>(degree));
+    basisPtr = Teuchos::rcp(new Intrepid2::Basis_HVOL_LINE_Cn_FEM<host_device_type, value_type, value_type>(degree));
     basisPtr->getDofCoeffs(Kokkos::subview(subcellTangents, Kokkos::ALL(),0));
   } else if (subcellBaseKey == shards::Triangle<>::key) {
-    basisPtr = Teuchos::rcp(new Intrepid2::Basis_HCURL_TRI_In_FEM<ExecutionSpace, ScalarType, ScalarType>(degree));
+    basisPtr = Teuchos::rcp(new Intrepid2::Basis_HCURL_TRI_In_FEM<host_device_type, value_type, value_type>(degree));
     basisPtr->getDofCoeffs(subcellTangents);
   } else if (subcellBaseKey == shards::Quadrilateral<>::key) {
-    basisPtr =  Teuchos::rcp(new Intrepid2::Basis_HCURL_QUAD_In_FEM<ExecutionSpace, ScalarType, ScalarType>(degree));
+    basisPtr =  Teuchos::rcp(new Intrepid2::Basis_HCURL_QUAD_In_FEM<host_device_type, value_type, value_type>(degree));
     basisPtr->getDofCoeffs(subcellTangents);
   }
 
   // coordinates \xi_j
-  ScalarViewType subcellDofCoords("subcellDofCoords", basisPtr->getCardinality(), subcellDim);
+  Kokkos::DynRankView<value_type,host_device_type> subcellDofCoords("subcellDofCoords", basisPtr->getCardinality(), subcellDim);
   basisPtr->getDofCoords(subcellDofCoords);
   INTREPID2_TEST_FOR_EXCEPTION( basisPtr->getDofCount(subcellDim,0) != ndofSubcell,
       std::logic_error,
       ">>> ERROR (Intrepid::OrientationTools::getCoeffMatrix_HCURL): " \
       "the number of basisPtr internal DoFs should equate those of the subcell");
 
   // restrict \xi_j (and corresponding t_j) to points internal to the HCURL basis
-  PointViewType refPtsSubcell("refPtsSubcell", ndofSubcell, subcellDim);
-  ScalarViewType refSubcellTangents("subcellTangents", ndofSubcell, subcellDim);
-  auto tagToOrdinal = Kokkos::create_mirror_view_and_copy(typename ExecutionSpace::memory_space(), basisPtr->getAllDofOrdinal());
-  Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>(0, ndofSubcell),
-      KOKKOS_LAMBDA (const int i){
+  Kokkos::DynRankView<value_type,host_device_type> refPtsSubcell("refPtsSubcell", ndofSubcell, subcellDim);
+  Kokkos::DynRankView<value_type,host_device_type> refSubcellTangents("subcellTangents", ndofSubcell, subcellDim);
+  auto tagToOrdinal = basisPtr->getAllDofOrdinal();
+  for (ordinal_type i=0;i<ndofSubcell;++i) {
```

Comment:
@kyungjoo-kim, is there any reason not to keep a parallel_for here, with `Kokkos::RangePolicy<typename host_device_type::ExecutionSpace>`?  (The advantage is that OpenMP would get to do this in parallel.)

Reply:
I do not think that it can gain performance from parallel operation even for the openmp case.  
1. the workload is quite small as the operation is just filling the data for a single matrix. 
2. in my opinion, sharing the work with multiple threads just cause more overhead as it can share the data on different caches losing locality.

Reply:
@kyungjoo-kim, OK; thanks for clarifying.  That sounds good to me.